sql-data-pipeline/src/main/java/io/cdap/pipeline/sql/app/bigquery/BigQueryExecutor.java

```
@@ -0,0 +1,122 @@
+    this.job = job;
+    this.serviceAccountPath = serviceAccountPath;
+  }
+
+  @Override
+  public void configure(CustomActionConfigurer configurer) {
+    Gson gson = new GsonBuilder()
+      .registerTypeAdapter(Queryable.class, new QueryableAdapter())
+      .registerTypeAdapter(Operand.class, new OperandAdapter())
+      .registerTypeAdapter(Constant.class, new ConstantAdapter())
+      .create();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(JOB_NAME, gson.toJson(job));
+    properties.put(SERVICE_ACCOUNT_PATH_NAME, serviceAccountPath);
+    configurer.setProperties(properties);
+  }
+
+  @Override
+  public void initialize(CustomActionContext context) {
+    Gson gson = new GsonBuilder()
+      .registerTypeAdapter(Queryable.class, new QueryableAdapter())
+      .registerTypeAdapter(Operand.class, new OperandAdapter())
+      .registerTypeAdapter(Constant.class, new ConstantAdapter())
+      .create();
+    job = gson.fromJson(context.getSpecification().getProperty(JOB_NAME), DefaultSQLJob.class);
+    serviceAccountPath = context.getSpecification().getProperty(SERVICE_ACCOUNT_PATH_NAME);
+  }
+
+  @Override
+  public void run() throws Exception {
+    StringBuilder queryBuilder = new StringBuilder();
+    BigQuerySQLSerializer serializer = new BigQuerySQLSerializer();
+    // Construct queries
+    for (StructuredStatement action: job.getActions()) {
+      if (action.getQuery() == null) {
+        throw new IllegalArgumentException("Unable to execute statement without a query.");
+      }
+      if (action.getTemporaryTable() != null) {
+        queryBuilder.append(serializer.getSQL(action.getQuery(), action.getTemporaryTable()));
+      } else if (action.getInsertIntoTable() != null) {
+        queryBuilder.append(serializer.getInsertIntoSQL(action.getQuery(), action.getInsertIntoTable()));
+      } else {
+        throw new IllegalArgumentException("Statement must contain a table declaration (for now).");
```

remove '(for now)'.

Also, these types of errors should cause things to fail early at app deployment, not later at runtime.

---

Fixed, and moved the check and exception into the configure method